spec/services/support_interface/revert_rejection_spec.rb

```diff
@@ -0,0 +1,18 @@
+require 'rails_helper'
+
+RSpec.describe SupportInterface::RevertRejection, with_audited: true do
+  describe '#save!' do
+    it 'reverts the application choice status back to `awaiting_provider_decision` and sets an audit comment' do
+      application_choice = create(:application_choice, :with_rejection)
+      zendesk_ticket = 'becomingateacher.zendesk.com/agent/tickets/example'
+
+      described_class.new(
+        application_choice: application_choice,
+        zendesk_ticket: zendesk_ticket,
+      ).save!
+
+      expect(application_choice).to be_awaiting_provider_decision
+      expect(application_choice.audits.last.comment).to include(zendesk_ticket)
+    end
+  end
+end
```

Comment:
Quite liked the testing pattern used [here](https://github.com/DFE-Digital/apply-for-teacher-training/pull/4572/files#diff-6e3f92ab4664c6b3c021cb2511b292d4a484c9c4acec2bde45f3ba7f1dfda083) where DeclineOffer is used to make the state change and we expect equality between the original choice and the updated one. Would it make sense to do similar here?